src/Blocks/BlockButton.h

```diff
@@ -45,18 +46,10 @@ class cBlockButtonHandler :
 		a_WorldInterface.GetBroadcastManager().BroadcastSoundEffect("block.stone_button.click_on", SoundPos, 0.5f, 0.6f);
 
 		// Queue a button reset (unpress)
-		auto TickDelay = (m_BlockType == E_BLOCK_STONE_BUTTON) ? 20 : 30;
-		a_Player.GetWorld()->ScheduleTask(TickDelay, [SoundPos, Pos, this](cWorld & a_World)
-			{
-				if (a_World.GetBlock(Pos) == m_BlockType)
-				{
-					// Block hasn't change in the meantime; set its meta
-					a_World.SetBlockMeta(Pos.x, Pos.y, Pos.z, a_World.GetBlockMeta(Pos) & 0x07, false);
-					a_World.WakeUpSimulators(Pos);
-					a_World.BroadcastSoundEffect("block.stone_button.click_off", SoundPos, 0.5f, 0.5f);
-				}
-			}
-		);
+		auto TickDelay = (m_BlockType == E_BLOCK_STONE_BUTTON) ? 10 : 15;
+
+		auto ChunkData = static_cast<cIncrementalRedstoneSimulator *>(a_Player.GetWorld()->GetRedstoneSimulator())->GetChunkData();
```

Comment:
You can't assume this is an instance of `cIncrementalRedstoneSimualtor`. It could be `cRedstoneNoopSimulator` instead, for example.

Reply:
True...but it is done many more times in the project.

Reply:
What is the 'cRedstoneNoopSimulator' used for?

Reply:
>True...but it is done many more times in the project.

It's used inside `cIncrementalRedstoneSimualtor` itself, where the fact that it's being called at all lets it know that it's the active simulator. So in there, it's always valid.

>What is the 'cRedstoneNoopSimulator' used for?

It replaces the redstone simulator with "no operation", i.e. it disables redstone entirely. That can help with performance or debugging for example.

Reply:
I see...so if redstone is disabled entirely then, I can just dynamic cast it and use the MechanismDelays like I need/wanted to. Because if redstone is disabled, buttons don't need to get triggered at all - right? :)

Reply:
I could use the version of handling from before I changed it for the noop-simulator...to visually press the button, right?

Reply:
Could you move all delay logic into the redstone handler?

Reply:
The way I tried to access the redstone handler from the OnUse-function is not correct like I did it. Can you give me a hint?

Reply:
Could you just set the button on and let the redstone simulator update do the rest?

@tigerw would know best. But I think when you find the button on and find  `DelayInfo == nullptr` then you should set the initial delay there.

Reply:
Okay

Reply:
But I will (have to) distinquish between noop and incremental mode - because in noop the button would not come back, if the update would handle the delays

Reply:
Which means I could just set the delay initially on button press... @tigerw - what would be best here? I will push and show what I mean with "distinquish"